components/layout_2020/geom.rs

```diff
@@ -522,3 +538,37 @@ impl From<LogicalRect<CSSPixelLength>> for LogicalRect<Au> {
         }
     }
 }
+
+pub(crate) trait ToLogical<Unit, LogicalType> {
+    fn to_logical(&self, writing_mode: WritingMode) -> LogicalType;
+}
+
+impl<Unit: Clone + Copy> ToLogical<Unit, LogicalRect<Unit>> for PhysicalRect<Unit> {
```

Comment:
I think the `Clone` bound is not required here as `Copy` implies `Clone`.

Reply:
Great point. I've removed `Clone` here.